pkg/querier/ingester_querier.go

```diff
@@ -94,36 +99,36 @@ func (q *IngesterQuerier) forAllIngesters(ctx context.Context, f func(context.Co
 		if err != nil {
 			return nil, err
 		}
-		return q.forGivenIngesterSets(ctx, replicationSets, f)
+		return q.forGivenIngesterSets(ctx, waitForAllResponses, replicationSets, f)
 	}
 
 	replicationSet, err := q.ring.GetReplicationSetForOperation(ring.Read)
 	if err != nil {
 		return nil, err
 	}
 
-	return q.forGivenIngesters(ctx, replicationSet, defaultQuorumConfig(), f)
+	return q.forGivenIngesters(ctx, replicationSet, defaultQuorumConfig, f)
 }
 
 // forGivenIngesterSets runs f, in parallel, for given ingester sets
-func (q *IngesterQuerier) forGivenIngesterSets(ctx context.Context, replicationSet []ring.ReplicationSet, f func(context.Context, logproto.QuerierClient) (interface{}, error)) ([]responseFromIngesters, error) {
-	// Enable minimize requests so we initially query a single ingester per replication set, as each replication-set is one partition.
+// waitForAllResponses param can be used to require results from all ingesters in all replication sets. If this is set to false, the call will return as soon as we have a quorum by zone.
```

Comment:
Same here. Let's see if we can improve this comment. If I was unfamiliar with how this works, I would be surprised because `results from all ingesters in all replication sets` contradicts `quorum by zone` which by definition is a majority, not all.